sugar/lib/src/time/zone/providers/universal/universal_provider.dart

```diff
@@ -0,0 +1,707 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+import 'dart:collection';
+
+import 'package:b/b.dart';
+import 'package:collection/collection.dart';
+import 'package:meta/meta.dart';
+
+import 'package:sugar/src/time/offset.dart';
+import 'package:sugar/src/time/temporal_unit.dart';
+import 'package:sugar/src/time/zone/timezone.dart';
+
+part 'universal_provider.g.dart';
+
+/// A [Timezone] provider for the universal timezone database.
+///
+/// This provider uses a bundled timezone database to provide timezone
+/// information for all known timezones.
+class UniversalTimezoneProvider extends UnmodifiableMapBase<String, Timezone> {
```

Comment:
Nitpick: I think `EmbeddedTimezoneProvider` better represents what this class does.

Reply:
100%